File: frontend/src/lib/projectWorklist.ts
import { apiUrl } from "@/lib/api";

export type ProjectWorklistStatus = "generating" | "needs_review" | "completed" | "failed";

export interface ProjectWorklistItem {
  project_id: string;
  project_name: string;
  status: ProjectWorklistStatus;
  thumbnail_url: string | null;
  result_url: string | null;
  review_url: string | null;
  export_history_url: string;
  last_export_status: string | null;
  updated_at: string;
}

interface ProjectWorklistResponse {
  items: ProjectWorklistItem[];
}

function sessionHeaders(): Record<string, string> { return {}; }

export async function fetchProjectWorklist(): Promise<ProjectWorklistItem[]> {
  const response = await fetch(apiUrl("/api/v1/projects/worklist"), {
    headers: sessionHeaders(), credentials: "include",
    cache: "no-store",
  });
  if (!response.ok) {
    throw new Error("작업 목록을 불러오지 못했습니다.");
  }
  const body: ProjectWorklistResponse = await response.json();
  return body.items;
}

File: backend/src/schemas/project_worklist.py
from pydantic import BaseModel


class ProjectWorklistItem(BaseModel):
    project_id: str
    project_name: str
    status: str
    thumbnail_url: str | None = None
    result_url: str | None = None
    review_url: str | None = None
    export_history_url: str
    last_export_status: str | None = None
    run_id: str | None = None
    updated_at: str


class ProjectWorklistResponse(BaseModel):
    items: list[ProjectWorklistItem]
